sn_node/src/put_validation.rs

```diff
@@ -233,6 +236,18 @@ impl Node {
             }
         };
 
+        // Notify the sender of any double spend
+        if validated_spends.len() > 1 {
+            warn!("Got a double spend for the SpendDbc PUT with dbc_id {dbc_id:?}",);
+            let mut proof = validated_spends.iter();
+            if let (Some(spend_one), Some(spend_two)) = (proof.next(), proof.next()) {
+                return Err(ProtocolError::DoubleSpendAttempt(
+                    Box::new(spend_one.to_owned()),
+                    Box::new(spend_two.to_owned()),
+                ))?;
+            }
+        }
+
```

Comment:
If we were to move this check to here, we would not be storing the double spend right?
Assuming, local copy had 1 spend and we got a PUT with another spend for the same `DbcId`, the above `signed_spend_validation` would aggregate those two and would produce a `Vec<SignedSpend>` with `len = 2`. We might want to store that to the network?